### DISCUSSION TITLE
Is There A Reason Why PKHeX.Drawing Is Split Up So Much?

### DISCUSSION THREAD

**Discussion by foohyfooh** [kwsch/PKHeX#4039] [Q&A] [answered]
PKHeX.Drawing is basically a small set of functions which is then consumed by PKHeX.Drawing.PokeSprite which is then further consumed by PKHeX.Drawing.Misc and then all three are consumed by PKHeX.WinForms. But if they are all consumed the main application why even have them like this instead of a single project in the solution?

**Comment by mgibson041387** [comment]
If you want to build it to a single file use the portion of the guide labeled "Building as a Single File" located towards the bottom. I have included the link below.

https://github.com/kwsch/SysBot.NET/wiki/Building-the-.sln

**Reply by foohyfooh** [reply]
It is not that I was looking for a single file but more I was trying to understand the design decision.

**Reply by mgibson041387** [reply]
If I had to guess I would say the misc.all file is possibly for items. Sprite.dll deals with the Pokémon images. Core.dll keeps track of Pokemon event data. I'm pretty sure these files all exist separately for cleanliness and organization.

**Comment by sora10pls** [comment, marked as answer]
Some reasoning was provided in the commit that split the project up in the first place: https://github.com/kwsch/PKHeX/commit/bc809af06cb4743bf8057960b14fd34fd8a49914

Some projects, like [PoGoEncTool](https://github.com/projectpokemon/PoGoEncTool), don't depend on all of the PKHeX.Drawing projects.
[pkNX](https://github.com/kwsch/pkNX) was also eventually changed to depend on PKHeX.Drawing.PokeSprite, since some sprites included in PKHeX.Drawing.Misc, like the box wallpapers and ribbon sprites, are not necessary for the tool.
